### DISCUSSION TITLE
Why is a upgrade uninstalling the old version first? Is this expected? Can it be avoided?

### DISCUSSION THREAD

**Discussion by sgon00** [Homebrew/discussions#1296] [Getting started]
Hi, I am new to brew in MacOS and come from Linux world.
Today, when I try to upgrade some apps (`brew upgrade app`), I fount out brew uninstalled them first.
This causes my config files of those apps are deleted too.
I can complain those apps author should not delete config files when uninstalling, BUT why uninstalling at the first place?
In linux, upgrade is just an upgrade. I might be wrong, but I never saw uninstalling behavior before.
Is this uninstalling behavior upon upgrade expected? Can it be avoided?
Thank you very much.

**Comment by SMillerDev** [comment]
It's this a cask or a formula?

**Reply by sgon00** [reply]
> It's this a cask or a formula?

Thanks a lot for your reply. I just check some of apps I tried to upgrade. I think they are casks although they are open source. They are in `/usr/local/Caskroom/` path.

**Reply by SMillerDev** [reply]
In that case the answer is very simply, you can't have the same app installed twice. So before it runs the installer again with the new version it needs to get rid of the old one.

**Reply by sgon00** [reply]
Thanks. I never really realized they are casks before you asked. I just don't get why app configurations are gone after the uninstallation. It's like a fresh install. I have to setup everything again which is not what I expect. Anyway, maybe I have to live with it. Thanks.

**Reply by SMillerDev** [reply]
@Homebrew/cask might have a way to mark some files to be retained between updates. But in the end this is a configuration in the upstream installer.

**Reply by sgon00** [reply]
Got it, Thank you very much for your help and answer.

**Comment by Misaka13514** [comment]
I've asked this question as well.
https://github.com/Homebrew/discussions/discussions/1205
